Add LockstakeMigrator (#65)
makerdao/lockstake: README.md

The StakingRewards contract `setRewardsDuration` function was modified to enable governance to change the farming distribution duration even if the previous distribution has not finished. This now supports changing it simultaneously with the SBE cooldown period (through a governance spell).

## 7. LockstakeMigrator

A contract which has the purpose to move `urn`s from a deprecated Lockstake version to a newer one, without having to pay the `exit` fee which would be required if the user would want to do this manually via the regular functions.
This contract uses the `LockstakeEngine.freeNoFee` function ensuring the collateral will still remain locked in a `LockstakeEngine`.
The migrator requires to be added to the `wards` mapping of the old `LockstakeEngine`.

There are two paths that the user could take when calling the `migrate` function for the desired `urn`:

> - If the `urn` doesn't have any debt. This is the simplest path where the collateral is just `free`d from the old engine and `locke`d in the new one.

Reviewer: ```suggestion
- If the `urn` doesn't have any debt. This is the simplest path where the collateral is just `free`d from the old engine and `lock`ed in the new one.
```

Author: Fixed!

The first path requires the migrator to be `hope`d in the old Engine for the `urn` being migrated. An authed address needs to call this `hope` function previously. It is also required that the caller of `migrate` be an authed address in the `urn` being migrated.
For the second path, apart from the same requirements of the simplest one, it is also necessary that an authed address in the `urn` that is receiving the position in the new Lockstake has `hope`d the migrator.
Also, the `migrate` executor needs to be an authed address in that recipient `urn`.

Note: the second path can encounter reverts on its execution, for example restrictions in the `line` of the `ilk` for the new Engine, or `dust` configuration between the two engines `ilk`s. So it might happen, that for a specific `urn` migration, it could be required that the user needs to manually repay totally or partially the debt or might even require to need to generate new debt for the migration to succeed.

> There might be other cases a part from these previous examples that could block a specific migration, the important thing to consider is that repaying totally the debt in a manual manner should always solve these issues.

Reviewer: ```suggestion
There might be other cases a part from these previous examples that could block a specific migration, the important thing to consider is that this is just a utility, so repaying totally the debt in a manual manner should always solve these issues.
```
Made a small addition, since I don't want it to sound like repaying the debt manually is the actual cure for all issues that might cause revert, since some other examples here are the old chief flash loan protection blocking a free, or someone sandwiching the call and temporarily taking the debt ceiling. For those, even doing the migration manually can cause a revert, and there are other mitigations (e.g using reserveHatch, flashbots, etc..).

Author: Added

Note 2: Migration won't transfer the `VoteDelegate` nor the farm selected in the old `urn` to the destination one. This needs to be manually done by an `urn` authed user directly in the new Engine (before or after the migration).

> Note 3: Migrator assumes `MkrSky` is configured without a penalty. So as soon as, the penalty is set above 0, the migrator will stop working.

Reviewer: ```suggestion
Note 3: Migrator assumes `MkrSky` is configured without a penalty. So as soon as, the penalty is set above 0, the migrator will generally stop working.
```
Made it a bit less committing, as if someone donates SKY to the migrator it might keep working even if there is a penalty.

Author: Added
